integration/experimental/v3_unshare_service_command_test.go

```diff
@@ -0,0 +1,319 @@
+				BeforeEach(func() {
+					session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName, "-o", sharedToOrgName)
+					Eventually(session).Should(Exit(0))
+				})
+
+				Context("when the org I want to unshare from does not exist", func() {
+					It("fails with an org not found error", func() {
+						session := helpers.CF("v3-unshare-service", serviceInstance, "-s", sharedToSpaceName, "-o", "missing-org")
+						Eventually(session).Should(Say("FAILED"))
+						Eventually(session.Err).Should(Say("Failed to unshare service instance '%s'. Ensure the space and specified org exist and that the service instance has been shared to this space.", serviceInstance))
+						Eventually(session).Should(Exit(1))
+					})
+				})
+
+				Context("when the space I want to unshare from does not exist", func() {
+					It("fails with a space not found error", func() {
+						session := helpers.CF("v3-unshare-service", serviceInstance, "-s", "missing-space", "-o", sharedToOrgName)
+						Eventually(session).Should(Say("FAILED"))
+						Eventually(session.Err).Should(Say("Failed to unshare service instance '%s'. Ensure the space and specified org exist and that the service instance has been shared to this space.", serviceInstance))
+						Eventually(session).Should(Exit(1))
+					})
+				})
+
+				Context("when I want to unshare my service instance from a space and org", func() {
+					It("successfully unshares the service instance", func() {
+						session := helpers.CF("v3-unshare-service", serviceInstance, "-s", sharedToSpaceName, "-o", sharedToOrgName)
+						Eventually(session).Should(Say("OK"))
+						Eventually(session).Should(Exit(0))
+					})
+				})
+			})
+
+			Context("when I have shared my service instance to a space within the targeted org", func() {
+				BeforeEach(func() {
+					helpers.CreateSpace(sharedToSpaceName)
+
+					session := helpers.CF("v3-share-service", serviceInstance, "-s", sharedToSpaceName)
+					Eventually(session).Should(Exit(0))
+				})
+
+				Context("when the space I want to unshare from does not exist", func() {
+					It("fails with a space not found error", func() {
+						session := helpers.CF("v3-unshare-service", serviceInstance, "-s", "missing-space")
+						Eventually(session).Should(Say("FAILED"))
+						Eventually(session.Err).Should(Say("Failed to unshare service instance '%s'. Ensure the space and specified org exist and that the service instance has been shared to this space.", serviceInstance))
+						Eventually(session).Should(Exit(1))
+					})
+				})
+
+				Context("when I want to unshare my service instance from the space", func() {
+					It("successfully unshares the service instance when I am admin", func() {
+						session := helpers.CF("v3-unshare-service", serviceInstance, "-s", sharedToSpaceName)
+						Eventually(session).Should(Say("OK"))
+						Eventually(session).Should(Exit(0))
+					})
+
+					Context("when I have no access to the shared-to space", func() {
```

Comment:
should this say when I only have read access?

Reply:
Hey @nickwei84 

To share a service the user must have SpaceDeveloper permissions in the source space & read permission in the target space. However, the desired behaviour is that users can unshare service instance when they do not have any access to the shared-to space. 

This is so SpaceDevelopers in the source space maintain complete control of that instance. So that for example, service instances cannot be held hostage in shared-to spaces. 

Sapi, @Samze + Alex

Reply:
Cool thanks!